src/app/modules/issue-vc/new-issue-vc/new-issue-vc.component.ts

```diff
@@ -21,175 +19,82 @@ export class NewIssueVcComponent implements OnInit {
     subject: ['', [Validators.required, HexValidators.isDidValid()]],
     type: ['', [Validators.required]]
   });
-  roles;
-  selectedRole;
-  selectedNamespace;
+  possibleRolesToEnrol;
+  selectedRoleDefinition;
   isPrecheckSuccess;
-  alreadyEnroled;
-  rolePreconditionList = [];
+  rolePreconditionList: PreconditionCheck[] = [];
 
   constructor(private fb: FormBuilder,
               @Inject(MAT_DIALOG_DATA) public data: { did: string },
               public dialogRef: MatDialogRef<NewIssueVcComponent>,
-              private loadingService: LoadingService,
-              private iamService: IamService,
               private issuanceVcService: IssuanceVcService) {
   }
 
   ngOnInit(): void {
-    this.issuanceVcService.getIssuerRoles().subscribe((roles) => {
-      this.roles = roles;
-    });
     this.setDid();
-
-    this.form.get('subject').valueChanges.subscribe(async (d) => {
-      if (this.form.get('subject').valid) {
-        await this.getNotEnrolledRoles(d);
-      }
-
-    });
+    this.setPossibleRoles();
   }
 
-  async roleTypeSelected(e: any) {
+  roleTypeSelected(e: any) {
```

Comment:
I would try to avoid using "any".  You should be able to see the type or just console.log(e) and see it there.